### ISSUE TITLE
iplan 1.9.0 uses CPU resources excessively

### ISSUE THREAD

**Issue by SokolovskyMaxim** [iman-salmani/iplan#49]
[Запись экрана от 2023-07-15 13-23-08.webm](https://github.com/iman-salmani/iplan/assets/106545001/4ed12eee-ce0c-4b36-ba25-2e945e13cfdc)

as can be seen from the screencast, excessive consumption of processor resources occurs during the operation of the application and stops when it is completed

It looks like a memory leak or something like that

**Comment by SokolovskyMaxim** [comment]
there are no lags when running through the terminal

**Comment by iman-salmani** [comment]
Thank you for reporting this
I tested the previous version and clearly something is wrong.

If you wanna downgrade to test or use the previous version:
```bash
flatpak update --commit=99bd4408cdbd65c95d18029e050802f4e769a5572d5b78f3eaaa6692d2fba53f ir.imansalmani.IPlan
```

Do you notice huge CPU usage change while doing specific task inside the app?

> there are no lags

Do you mean logs?

**Comment by SokolovskyMaxim** [comment]
Yes, the previous version does not have such a bug.

And yes, I meant logs

**Comment by HeadPlug** [comment]
This happens to me as well, on 1.9.2/Fedora 38
![Screenshot from 2023-07-19 14-19-28](https://github.com/iman-salmani/iplan/assets/57101103/11b3a188-21d5-4c86-a5f4-73801c1b778e)

**Comment by iman-salmani** [comment]
Hi @HeadPlug
Do you have this problem even after navigating to the calendar page and back to a project page?

**Comment by HeadPlug** [comment]
> At the first launch after the corrective update, I also had manifestations of the old error. But then, after restarting the application, the problem disappears and the error is not reproduced

I've restarted both the application and my computer multiple times; it keeps happening.

> Hi @HeadPlug Do you have this problem even after navigating to the calendar page and back to a project page?

No. After visiting Calendar, the issue solves itself. If I close IPlan and open it again, the issue will reoccur, and then I have to go to Calendar again. This happens even if I kill the iplan process. If I open IPlan from the terminal though, the issue doesn't happen.

[Kooha-2023-07-20-04-24-43.webm](https://github.com/iman-salmani/iplan/assets/57101103/23e9d0c0-af10-4f33-81dd-0da19e209ac2)

**Comment by SokolovskyMaxim** [comment]
> I've restarted both the application and my computer multiple times; it keeps happening.

You're right, the error occurs when running through the application menu or through the search. I have a launch assigned to a keyboard shortcut, this is equivalent to launching from the terminal, so I did not notice that the error continues to be reproduced in extreme cases

**Comment by HeadPlug** [comment]
I've cracked it, I think - it's the .desktop file. 

I edited it down to this, and the issue does not happen anymore:
![Screenshot from 2023-07-20 12-16-23](https://github.com/iman-salmani/iplan/assets/57101103/74a71883-facb-4a23-8e26-fa1ebc745925)
I was however in a rush, so I don't remember what I deleted, but it should be fairly trivial to troubleshoot now.

**Comment by HeadPlug** [comment]
If a .desktop misconfig is capable of spiking the CPU so much, though, it might also be triggering some unknown bug in GTK4/GNOME/libAdwaita?

**Comment by HeadPlug** [comment]
Issue is resolved when changing `Name `to `Name[en]=IPlan ` instead of `Name=IPlan`

**Comment by muelsyse-az** [comment]
Just want to give an update on this issue. 
I was using the app and realized that my battery usage went up significantly and my fans started ramping. Confused, I went to look at my System Monitoring Center and found out that iPlan was using ~20% CPU when run. I am on iPlan 1.9.2 running Fedora Silverblue 38.

Since I have a rather small battery (40Wh), this makes it unfeasible for me to use the app when on battery (which I often do for prolonged durations due to school). I hope this is fixed, but I understand that this may take a while.

**Comment by niktss** [comment]
> Just want to give an update on this issue. I was using the app and realized that my battery usage went up significantly and my fans started ramping. Confused, I went to look at my System Monitoring Center and found out that iPlan was using ~20% CPU when run. I am on iPlan 1.9.2 running Fedora Silverblue 38.

Until this is fixed, you can use the workaround discussed above - simply go to the Calendar tab after every app launch, and CPU usage should normalise itself.

**Comment by muelsyse-az** [comment]
@niktss Thanks! I'll give that a go.
